### PR TITLE
apiextensions: avoid two HA API servers to fight for NonStructural condition message

### LINKED ISSUE #78060: CRD structural schema condition is susceptible to message changes
/sig api-machinery
/kind bug
/assign sttts

because the controller that maintains the condition runs multiple instances in HA apiserver clusters, the controller must not duel in cases where the validation messages change slightly between releases

### PR DESCRIPTION
**Pull request by sttts**
Fixes #78060

### CODE REVIEW

File: staging/src/k8s.io/apiextensions-apiserver/pkg/controller/nonstructuralschema/nonstructuralschema_controller.go [hunk 1]
```diff
@@ -130,6 +137,14 @@ func (c *ConditionController) sync(key string) error {
 		return err
 	}
 
+	// avoid repeated calculation for the same generation
+	c.lastSeenGenerationLock.Lock()
```
Comment (tedyu) [inline staging/src/k8s.io/apiextensions-apiserver/pkg/controller/nonstructuralschema/nonstructuralschema_controller.go]: It seems a read lock here should suffice.

Comment (liggitt) [reply staging/src/k8s.io/apiextensions-apiserver/pkg/controller/nonstructuralschema/nonstructuralschema_controller.go]: true, but this is a slow moving api with an extremely short critical section.  I don't expect significant contention here at all

Comment (tedyu) [reply staging/src/k8s.io/apiextensions-apiserver/pkg/controller/nonstructuralschema/nonstructuralschema_controller.go]: Thanks for the clarification.

**Pull request merged**